src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyPropertyGenerator.cs

```diff
@@ -25,7 +25,10 @@ public void Initialize(GeneratorInitializationContext context)
 
 		public void Execute(GeneratorExecutionContext context)
 		{
-			if (PlatformHelper.IsValidPlatform(context))
+			if (PlatformHelper.IsValidPlatform(context)
+
+				// This generator is only valid inside of Uno.UI
+				&& (context.Compilation.AssemblyName?.Equals("Uno.UI", StringComparison.OrdinalIgnoreCase) ?? false))
```

Comment:
I think this could use `context.GetMSBuildPropertyValue("_IsUnoUISolution") == "true"` to not depend on a specific assembly name.

https://github.com/unoplatform/uno/blob/4666c66d75ea47c9156ea79c5d1e83465fdf570d/src/Directory.Build.props#L145-L151

Or if https://github.com/unoplatform/uno/pull/9729 got ready fast enough, we will not need the check.

Reply:
That's fair, your change may be fast enough for this not to be needed at all :)